server/src/main/java/org/elasticsearch/action/admin/indices/create/AutoCreateAction.java

```diff
@@ -141,12 +152,56 @@ public ClusterState execute(ClusterState currentState) throws Exception {
                             return currentState;
                         }
 
-                        CreateIndexClusterStateUpdateRequest updateRequest =
-                            new CreateIndexClusterStateUpdateRequest(request.cause(), indexName, request.index())
-                                .ackTimeout(request.timeout()).masterNodeTimeout(request.masterNodeTimeout());
+                        CreateIndexClusterStateUpdateRequest updateRequest = buildUpdateRequest(indexName);
+
                         return createIndexService.applyCreateIndexRequest(currentState, updateRequest, false);
                     }
                 }
+
+                private CreateIndexClusterStateUpdateRequest buildUpdateRequest(String indexName) {
+                    boolean isSystemIndex = false;
+                    String mappings = null;
+                    Settings settings = null;
+                    String aliasName = null;
+                    String concreteIndexName = indexName;
+
+                    final SystemIndexDescriptor descriptor = systemIndices.findMatchingDescriptor(indexName);
+
+                    if (descriptor != null && descriptor.isAutomaticallyManaged()) {
```

Comment:
Can we simplify this method a bit? We don't really share logic between non-system index creation and system index creation as far as I can see so I think we can test for the non system case and simply return the update request there?